remote/mattermost/remote.go

```diff
@@ -0,0 +1,177 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package mattermost
+
+import (
+	"encoding/json"
+	"strconv"
+
+	"github.com/mattermost/mattermost-server/v6/model"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+
+	"github.com/target/flottbot/models"
+	"github.com/target/flottbot/remote"
+)
+
+// Client struct.
+type Client struct {
+	Server string
+	Token  string
+	BotID  string
+}
+
+type mmLogger struct {
+	zerolog.Logger
+}
+
+func (l *mmLogger) Output(_ int, s string) error {
+	l.Logger.Info().Msg(s)
+	return nil
+}
+
+// validate that Client adheres to remote interface.
+var _ remote.Remote = (*Client)(nil)
+
+// instantiate a new mattermost client.
+func (c *Client) new() *model.Client4 {
+	log.Info().Msgf("%#v", c)
+	api := model.NewAPIv4Client("http://" + c.Server)
+	api.SetToken(c.Token)
+
+	return api
+}
+
+func (c *Client) Name() string { return "mattermost" }
+
+func (c *Client) Reaction(_ models.Message, rule models.Rule, _ *models.Bot) {
+	if rule.RemoveReaction != "" {
+		log.Debug().Msg("Remove reaction not implemented for mattermost")
+	}
+
+	if rule.Reaction != "" {
+		log.Debug().Msg("Reactions not implemented for mattermost")
+	}
+
+}
+
+func (c *Client) Read(inputMsgs chan<- models.Message, _ map[string]models.Rule, bot *models.Bot) {
+	api := c.new()
+	if user, resp, err := api.GetUser("me", ""); err != nil {
+		log.Fatal().Msgf("Could not login, %s", err)
+	} else {
+		log.Info().Interface("user", user.Username).Interface("resp", resp).Msg("")
+		log.Info().Msg("Logged in to mattermost")
+
+		c.BotID = user.Username
+	}
+
+	sock, err := model.NewWebSocketClient4("ws://"+c.Server, c.Token)
```

Comment:
Similar to previous comment, I presume this would be `wss:` for most installations?

Reply:
Yep I'll update that. It might be the issue I'm seeing with my work instance.